src/click-outside.directive.ts

```diff
@@ -1,67 +1,89 @@
-    this._initOnClickBody = this._initOnClickBody.bind(this);
-    this._onClickBody = this._onClickBody.bind(this);
-  }
+    constructor(
+        @Inject(DOCUMENT) private _document /*: HTMLDocument*/,
+        private _el: ElementRef) {
+        this._initOnClickBody = this._initOnClickBody.bind(this);
+        this._onClickBody = this._onClickBody.bind(this);
+    }
+
+    ngOnInit() {
+        this._init();
+    }
 
-  ngOnInit() {
-    this._init();
-  }
+    ngOnDestroy() {
+        if (this.attachOutsideOnClick) {
+            this._el.nativeElement.removeEventListener('click', this._initOnClickBody);
+        }
+
+        this._document.body.removeEventListener('click', this._onClickBody);
+    }
 
-  ngOnDestroy() {
-    if (this.attachOutsideOnClick) {
-      this._el.nativeElement.removeEventListener('click', this._initOnClickBody);
+    ngOnChanges(changes: SimpleChanges) {
+        if (changes['attachOutsideOnClick'] &&
+            changes['attachOutsideOnClick'].previousValue !== changes['attachOutsideOnClick'].currentValue) {
+            this._init();
+        }
     }
 
-    this._document.body.removeEventListener('click', this._onClickBody);
-  }
+    private _init() {
 
-  ngOnChanges(changes: SimpleChanges) {
-    if (changes['attachOutsideOnClick'] &&
-      changes['attachOutsideOnClick'].previousValue !== changes['attachOutsideOnClick'].currentValue) {
-      this._init();
+        let nodeExcluded = this.exclude.split(',');
+
+        nodeExcluded.forEach((selector) => {
+            let node = this._document.querySelector(selector);
+            if(node != null) this.nodeExcluded.push(node);
+        });
+
+        if (this.attachOutsideOnClick) {
+            this._el.nativeElement.addEventListener('click', this._initOnClickBody);
+        } else {
+            this._initOnClickBody();
+        }
     }
-  }
 
-  private _init() {
-    if (this.attachOutsideOnClick) {
-      this._el.nativeElement.addEventListener('click', this._initOnClickBody);
-    } else {
-      this._initOnClickBody();
+    private _initOnClickBody() {
+        this._document.body.addEventListener('click', this._onClickBody);
     }
-  }
 
-  private _initOnClickBody() {
-    this._document.body.addEventListener('click', this._onClickBody);
-  }
+    private _onClickBody(e: Event) {
+        if (!this._el.nativeElement.contains(e.target) && !this.isExclude(e.target)) {
+            this.clickOutside.emit(e);
 
-  private _onClickBody(e: Event) {
-    if (!this._el.nativeElement.contains(e.target)) {
-      this.clickOutside.emit(e);
+            if (this.attachOutsideOnClick) {
+                this._document.body.removeEventListener('click', this._onClickBody);
+            }
+        }
+    }
 
-      if (this.attachOutsideOnClick) {
-        this._document.body.removeEventListener('click', this._onClickBody);
-      }
+    private isExclude(target):boolean {
+        var flag = false;
+        this.nodeExcluded.forEach((node) => {
```

Comment:
Might be better to just use a regular forloop so we can prematurely break out of it when `flag` gets set to `true`. The `return` will only break off the one iteration but still continue on for the rest.